lib/fizzy/parser.cpp

```diff
@@ -116,10 +116,12 @@ inline parser_result<Global> parse(const uint8_t* pos, const uint8_t* end)
 template <>
 inline parser_result<Table> parse(const uint8_t* pos, const uint8_t* end)
 {
-    (void)end;  // FIXME: Bounds checking.
-    const uint8_t kind = *pos++;
-    if (kind != FuncRef)
-        throw parser_error{"unexpected table elemtype: " + std::to_string(kind)};
+    if (pos == end)
+        throw parser_error{"Unexpected EOF"};
+
+    const auto elemtype = *pos++;
```

Comment:
Why not keep the `uint8_t` type here or do a static_cast for explicitness? I know the input pointer is `uint8_t`.

Reply:
Changed.